src/components/Navigator/NavigatorCard.vue

```diff
@@ -245,17 +236,14 @@ export default {
       type: Boolean,
       default: false,
     },
-    enableQuickNavigation: {
-      type: Boolean,
-      default: false,
-    },
     allowHiding: {
       type: Boolean,
       default: true,
     },
   },
   mixins: [
     keyboardNavigation,
+    symbolTreeNavigator,
```

Comment:
I understand why you extracted those into a mixin, but now this component looks like its missing an important functionality. Mixins are hard to reason with, as they can hold hidden functionality, which is good and bad at the same time.

Since the navigator dataset is now used in other places, I think it makes more sense to extract all of these helper functions like `getParent`, `getSiblings` and so on into their own file, so they are widely available and easy to inject into other components, as long as they have access to the dataset.

Reply:
Thanks for pointing this out! I was extracting only the functions I needed on the other component, but it does make sense to extract all others (`getAllChildren` `getSiblings` `getChildren`) `getParents` is already on the file. Do you have any preference on how we should name this mixin?

Reply:
I generally am not a fan of Mixins, as they are hard to reason with. You could use a mixin on just export a bunch of methods from a file, and could call these `NavigatorIndexUtils`.

Reply:
I like the utils file approach, but since all the functions depend on a computed property, is this not an indicator that it should be converted into a mixin? I don't have a strong preference for either of them, just wondering since both seem to be very similar solutions

Reply:
Hmmm I see what you mean... Its more annoying to refactor those as props, so yeah, I think a mixin should be fine in that case.

Reply:
The refactor was straightforward https://github.com/apple/swift-docc-render/pull/396/commits/0e1f5cff7caf0fe87ec9fdec752ad3b33c72f192. We're getting rid of the mixin, but now each method requires extra parameters that before would be coming from the computed properties, both ways seem fine to me

Reply:
yeah those are fine i think.